src/Generator/ClassGenerator.php

```diff
@@ -39,41 +39,35 @@ class ClassGenerator extends AbstractGenerator implements TraitUsageInterface
     public const FLAG_ABSTRACT      = 0x01;
     public const FLAG_FINAL         = 0x02;
 
-    /** @var FileGenerator */
-    protected $containingFileGenerator;
+    protected ?FileGenerator $containingFileGenerator = null;
 
-    /** @var string */
-    protected $namespaceName;
+    protected ?string $namespaceName = null;
 
-    /** @var DocBlockGenerator */
-    protected $docBlock;
+    protected ?DocBlockGenerator $docBlock = null;
 
-    /** @var string */
-    protected $name;
+    protected string $name = '';
 
-    /** @var bool */
-    protected $flags = 0x00;
+    protected int $flags = 0x00;
 
-    /** @var string */
-    protected $extendedClass;
+    protected ?string $extendedClass = null;
 
     /**
      * @var string[] Array of string names
-     * @psalm-var array<class-string>
```

Comment:
For codegen purposes we often use string names from different sources, thus static check for `class-string` will mostly fail, as it is impossible to add additional guards like `class_exists`, etc.

So, I'm suggesting to use parent `string` type here. WDYT?

Reply:
IMO class-string to be kept: if symbol names are being invented, we declare them from the caller side, but in here we should work with strict guarantees.

Reply:
@Ocramius I also want to have strict guarantees. But don't want to add suppress or custom docBlocks in client code. 

Consider this code from the`ClassGeneratorTest`:

```php
        $classGenerator->setImplementedInterfaces([
            'SomeNamespace\ClassInterface',
            GeneratorInterface::class,
            'Iteratable',
        ]);
```

How would you like to handle `class-string` guarantee for 1st and 3rd arguments? Psalm checks that only `::class` values can be passed and if we don't have an interface which we want to implement (eg. interface is being generated), then this check will always fail.

Possible solutions:
1) Add `@psalm-suppress' to such calls. Not a good idea..
2) Extract a variable and add a `@psalm-var array<class-string>` docBlock with proper type before calling `setImplementedInterfaces()`. Aka "visual debt". From client's POV this will be strange again. 

Is it ok for you to force usage of type casting if necessary to the client side to make sure that all interfaces are present?

Reply:
> But don't want to add suppress or custom docBlocks in client code.

It's up to the client to ensure that the input is a valid class-alike symbol.

 > Aka "visual debt". From client's POV this will be strange again.

Not a problem either. We add types to push some validation constraints to the edge of the source (consumers), which is normal and happens every time types are being refined.

Reply:
Ok, accepted 👍